test/testRules.js

```diff
@@ -996,6 +997,192 @@ export const testRules = {
+                target: "{count, plural, one {Dies ist einzigartig} other {Dies ist mehrerartig}}",
+                pathName: "a/b/c.xliff",
+                state: "new"
+            }),
+            file: "x/y"
+        });
+        const expected = new Result({
+            severity: "error",
+            description: "Resources must have the following state: translated",
+            id: "plural.test",
+            highlight: 'Resource found with disallowed state: <e0>new</e0>',
+            rule,
+            pathName: "x/y",
+            locale: "de-DE",
+            source: '{count, plural, one {This is singular} other {This is plural}}'
+        });
+        test.deepEqual(actual, expected);
+
+        test.done();
+    },
+
+    testResourceStateCheckerMatchArrayError: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceStateChecker({
+            // all resources should have this state:
+            param: [ "translated", "needs-review" ]
+        });
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "plural.test",
+                sourceLocale: "en-US",
+                source: '{count, plural, one {This is singular} other {This is plural}}',
+                targetLocale: "de-DE",
+                target: "{count, plural, one {Dies ist einzigartig} other {Dies ist mehrerartig}}",
+                pathName: "a/b/c.xliff",
+                state: "new"
+            }),
+            file: "x/y"
+        });
+        const expected = new Result({
+            severity: "error",
+            description: "Resources must have one of the following states: translated, needs-review",
+            id: "plural.test",
+            highlight: 'Resource found with disallowed state: <e0>new</e0>',
+            rule,
+            pathName: "x/y",
+            locale: "de-DE",
+            source: '{count, plural, one {This is singular} other {This is plural}}'
+        });
+        test.deepEqual(actual, expected);
+
+        test.done();
+    },
+
+    testResourceStateCheckerMatchDefaultNoError: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceStateChecker();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "plural.test",
+                sourceLocale: "en-US",
+                source: '{count, plural, one {This is singular} other {This is plural}}',
+                targetLocale: "de-DE",
+                target: "{count, plural, one {Dies ist einzigartig} other {Dies ist mehrerartig}}",
+                pathName: "a/b/c.xliff",
+                state: "translated"
+            }),
+            file: "x/y"
+        });
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceStateCheckerMatchDefaultError: function(test) {
+        test.expect(2);
```

Comment:
Don't we need to add a test case with testing the actual xliff file?

Reply:
These tests and the others in this file assume that that the resource file is converted correctly into Resource instances. If so, then this Rule instance will work too.